go/adbc/driver/snowflake/statement.go

```diff
@@ -584,6 +584,41 @@ func (st *statement) ExecuteUpdate(ctx context.Context) (int64, error) {
 	return n, nil
 }
 
+// ExecuteQuery executes the current query or prepared statement
+// and returnes a RecordReader for the results along with the number
+// of rows affected if known, otherwise it will be -1.
+//
+// This invalidates any prior result sets on this statement.
```

Comment:
nit: docstring is out of sync (IMO, it's not necessary here since you would only use it through the interface)

Reply:
It felt inconsistent with the other functions not to include it (assuming, *ahem*, I'd remembered to edit it...).